### PR TITLE
[Audit Suggestion 6] Update dependencies

### PR DESCRIPTION
**Pull request by nkryuchkov**
Update dependencies to latest released versions reducing the amount of warnings returned by `go list -json -m all | nancy sleuth` and `go list -u -m -json all | go-mod-outdated`.

`github.com/libp2p/go-libp2p-core/peer` was deprecated and causes linter to fail, hence `github.com/libp2p/go-libp2p/core/peer` is used

### CODE REVIEW

**Comment by GalRogozinski** [comment]
Did you see #265?
I upgraded only what was easy to update.

We need to think how much time we should spend on this since spec isn't really production code.
Notice some dependencies are not really used by spec atm like libp2p (there is unfinished msg validation in spec atm).

**Comment by nkryuchkov** [comment]
@GalRogozinski 

> Did you see https://github.com/bloxapp/ssv-spec/pull/265?

Yes, but I would update other dependencies as well

> I upgraded only what was easy to update.

Did you face any complexities with upgrading the rest?

> We need to think how much time we should spend on this since spec isn't really production code.

The spec itself isn't but the implementation imports the spec, so some spec code is production code. I spent not much time on that and both linter/tests passed locally. Unfortunately, CI fails right now, I'll try to fix it quickly. If it will require time, I'll revert some upgrades

> Notice some dependencies are not really used by spec atm like libp2p (there is unfinished msg validation in spec atm).

The approach of starting/continuing things like msg validation with the new libp2p version sounds better to me than using an old version and then rewriting it later

File: types/spectest/generate/tests.json
Comment (nkryuchkov) [inline types/spectest/generate/tests.json:1]: This files gets changed after running `make generate-jsons` on `main` too, so

Comment (GalRogozinski) [reply types/spectest/generate/tests.json:1]: you can revert this...
We have a test that generates random keys so it walwyas changes the test

Comment (nkryuchkov) [reply types/spectest/generate/tests.json:1]: Oh, I didn't think about this. Thanks! Reverted

**Review by GalRogozinski — APPROVED** [review]
Good job

**Review by lior-blox — APPROVED** [review]

**Pull request merged**